### PR TITLE
Use stderr for warnings

### LINKED ISSUE #155: Print warnings and errors to stderr instead of stdout

### PR DESCRIPTION
**Pull request by mathias-baumann-sociomantic**
Fixes #155

### CODE REVIEW

File: src/lib/shell/helper.d [hunk 1]
```diff
@@ -285,7 +333,7 @@ void keepTrying ( alias printer = writefln ) ( void delegate ( ) dg,
     foreach (_; 0..max_attempts) try
     {
         if (exception !is null)
-            printer("retrying ...");
+            errstream.isNull || errstream.writefln("retrying ...");
```
Comment (leandro-lucarella-sociomantic) [inline src/lib/shell/helper.d]: Is an `if` so expensive? :)

Comment (mathias-baumann-sociomantic) [reply src/lib/shell/helper.d]: it felt... easier to read to me

**Comment by mathias-baumann-sociomantic** [comment]
> Is there any reason not to have some sort of global module defining printerr() and printwarn()` functions or something?

I didn't consider it useful..

**Comment by mathias-baumann-sociomantic** [comment]
Updated to use more `if`s to make lucas heart happy

File: src/lib/shell/helper.d [hunk 2]
```diff
@@ -297,18 +345,22 @@ void keepTrying ( alias printer = writefln ) ( void delegate ( ) dg,
...
+            errstream.writefln("failed: %s", exc.msg);
 
         Thread.sleep(wait_time);
     }
 
     if (exception !is null)
     {
-        printer("Giving up after %s attempts", max_attempts);
+        if (!errstream.isNull)
+            errstream.writefln("Giving up after %s attempts", max_attempts);
```
Comment (mihails-strasuns-sociomantic) [inline src/lib/shell/helper.d]: Just define new `tryWritefln` wrapper that does the null check.

**Comment by mathias-baumann-sociomantic** [comment]
added `tryErrWritefln`

File: src/lib/shell/helper.d [hunk 3]
```diff
@@ -271,21 +295,51 @@ public string readPassword ( string question )
...
+*******************************************************************************/
+
+void keepTrying ( void delegate ( ) dg, int max_attempts, Duration wait_time,
+    Nullable!File outstream, Nullable!File errstream )
 {
     import core.thread;
 
     Exception exception;
 
+    auto tryErrWritefln ( Args... ) ( Args args )
```
Comment (mihails-strasuns-sociomantic) [inline src/lib/shell/helper.d]: Why not generic `tryWritefln ( Args... ) ( Nullable!File stream, Args args )` ?

**Comment by mathias-baumann-sociomantic** [comment]
chaaaaanged again

**Pull request merged**